src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -185,47 +193,253 @@ bool SotaUptaneClient::initialize() {
     return false;
   }
 
-  uptane_repo.setPrimaryEcuSerialHwId(serials[0]);
+  uptane_manifest.setPrimaryEcuSerialHwId(serials[0]);
 
   return true;
 }
 
-void SotaUptaneClient::getUpdateRequests() {
-  reportNetworkInfo();
-  // Uptane step 1 (build the vehicle version manifest):
-  if (!putManifest()) {
-    return;
-  }
+bool SotaUptaneClient::updateDirectorMeta() {
   // Uptane step 2 (download time) is not implemented yet.
   // Uptane step 3 (download metadata)
-  if (!uptane_fetcher.fetchMeta()) {
-    LOG_ERROR << "could not retrieve metadata";
-    *events_channel << std::make_shared<event::UptaneTimestampUpdated>();
-    return;
+
+  // reset director repo to initial state before starting UPTANE iteration
+  director_repo.resetMeta();
```

Comment:
Why are we doing this?

Reply:
Defensive programming probably. I think we can move what `resetMeta()` does with `root` to `initRoot` and remove the rest. Kind of wanted to be sure that we start each iteration from a clean state.

Reply:
Or actually, maybe I'll just limit the lifetime of `director_repo` and `images_repo` to the scope of one iteration, it will look much better.